src/reducers/entities/posts.js

```diff
@@ -502,7 +502,8 @@ function handlePosts(posts = {}, postsInChannel = {}, postsInThread = {}, action
         return {posts, postsInChannel, postsInThread};
     case ChannelTypes.RECEIVED_CHANNEL_DELETED:
     case ChannelTypes.DELETE_CHANNEL_SUCCESS:
-        if (!action.data.viewArchivedChannels) {
+    case ChannelTypes.LEAVE_CHANNEL:
+        if (action.data && !action.data.viewArchivedChannels) {
             return handleChannelDeleted(posts, postsInChannel, postsInThread, action.data.id);
```

Comment:
nit: I think would be a good option to change the name of the function to `handleChannelDeletedOrLeft`

Reply:
Renamed to `handleChannelRemoved` to cover both deleted, left, and technically kicked.